test/integration/apiserver/flowcontrol/concurrency_util_test.go

```diff
@@ -0,0 +1,506 @@
+// increase request execution time to make the system noise relatively insignificant.
+// This test calculates the server-side observed concurrency from average priority level seat utilization APF metric.
+// It also assumes that
+// (server-side request execution throughput) == (client-side request throughput) and derives a formula to
+// calculate the client-side expected concurrency. The two are compared and a small error bound is determined
+// from estimating the noise using 2*(standard deviation of requenst latency)/(avg request latency).
+func TestConcurrencyIsolation(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, genericfeatures.APIPriorityAndFairness, true)()
+	// NOTE: disabling the feature should fail the test
+	// start webhook server
+	serv := &mockV1Service{allow: true, statusCode: 200}
+	s, err := NewV1TestServer(serv, testcerts.ServerCert, testcerts.ServerKey, testcerts.CACert)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer s.Close()
+
+	authorizer, err := newV1Authorizer(s.URL, testcerts.ClientCert, testcerts.ClientKey, testcerts.CACert, 0)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	kubeConfig, closeFn := setupWithAuthorizer(t, 10, 10, authorizer)
+	defer closeFn()
+
+	loopbackClient := clientset.NewForConfigOrDie(kubeConfig)
+	noxu1Client := getClientFor(kubeConfig, "noxu1")
+	noxu2Client := getClientFor(kubeConfig, "noxu2")
+
+	queueLength := 50
+	concurrencyShares := 100
+
+	plNoxu1, _, err := createPriorityLevelAndBindingFlowSchemaForUser(
+		loopbackClient, "noxu1", concurrencyShares, queueLength)
+	if err != nil {
+		t.Error(err)
+	}
+	plNoxu2, _, err := createPriorityLevelAndBindingFlowSchemaForUser(
+		loopbackClient, "noxu2", concurrencyShares, queueLength)
+	if err != nil {
+		t.Error(err)
+	}
+
+	stopCh := make(chan struct{})
+	wg := sync.WaitGroup{}
+
+	// "elephant"
+	noxu1NumGoroutines := 5 + queueLength
+	var noxu1LatMeasure clientLatencyMeasurement
+	wg.Add(noxu1NumGoroutines)
+	streamRequests(noxu1NumGoroutines, func() {
+		start := time.Now()
+		_, err := noxu1Client.CoreV1().Namespaces().Get(context.Background(), "default", metav1.GetOptions{})
+		duration := time.Since(start).Seconds()
+		noxu1LatMeasure.update(duration)
+		if err != nil {
+			t.Error(err)
+		}
+	}, &wg, stopCh)
+	// "mouse"
+	noxu2NumGoroutines := 3
+	var noxu2LatMeasure clientLatencyMeasurement
+	wg.Add(noxu2NumGoroutines)
+	streamRequests(noxu2NumGoroutines, func() {
+		start := time.Now()
+		_, err := noxu2Client.CoreV1().Namespaces().Get(context.Background(), "default", metav1.GetOptions{})
+		duration := time.Since(start).Seconds()
+		noxu2LatMeasure.update(duration)
+		if err != nil {
+			t.Error(err)
+		}
+	}, &wg, stopCh)
+
+	// Warm up
+	time.Sleep(testWarmUpTime)
+
+	noxu1LatMeasure.reset()
+	noxu2LatMeasure.reset()
+	// Snapshots maps from a time label to a metricSnapshot
+	snapshots := make(map[string]metricSnapshot)
```

Comment:
This outer layer of map structure is unnecessary complexity.
It would be more direct to just use two variables of type `metricSnapshot`, perhaps named `snapshot0` and `snapshot1`.